internal/databases/redis/delete_handler.go

```diff
@@ -0,0 +1,208 @@
+	if err != nil {
+		return err
+	}
+
+	_, _, err = HandleBackupsDelete(backupTimes, backupFolder, opts)
+	if err != nil {
+		return err
+	}
+
+	if opts.purgeGarbage {
+		tracelog.InfoLogger.Printf("Garbage prefixes in backups folder: %v", garbage)
+		if !opts.dryRun {
+			if err := DeleteGarbage(backupFolder, garbage); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// HandleBackupsDelete delete backups according to settings
+func HandleBackupsDelete(backupTimes []internal.BackupTime,
+	folder storage.Folder,
+	opts PurgeSettings) (purge, retain []archive.Backup, err error) {
+	if len(backupTimes) == 0 {
+		tracelog.InfoLogger.Println("No backups found")
+		return []archive.Backup{}, []archive.Backup{}, nil
+	}
+
+	backups, err := LoadBackups(folder, BackupNamesFromBackupTimes(backupTimes))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	timedBackup := archive.RedisModelToTimedBackup(backups)
+
+	purgeBackups, retainBackups, err := internal.SplitPurgingBackups(timedBackup, opts.retainCount, opts.retainAfter)
+	purge = archive.TimedBackupToRedisModel(purgeBackups)
+	retain = archive.TimedBackupToRedisModel(retainBackups)
+	if err != nil {
+		return nil, nil, err
+	}
+	tracelog.InfoLogger.Printf("Backups selected to be deleted: %v", BackupNamesFromBackups(purge))
+	tracelog.InfoLogger.Printf("Backups selected to be retained: %v", BackupNamesFromBackups(retain))
+
+	if !opts.dryRun {
+		if err := DeleteBackups(folder, purge); err != nil {
+			return nil, nil, err
+		}
+		tracelog.InfoLogger.Printf("Backups were purged: deleted: %d, retained: %v", len(purge), len(retain))
+	}
+	return purge, retain, nil
+}
+
+// LoadBackups downloads backups metadata
+func LoadBackups(folder storage.Folder, names []string) ([]archive.Backup, error) {
+	backups := make([]archive.Backup, 0, len(names))
+	for _, name := range names {
+		backup, err := BackupMeta(folder, name)
+		if err != nil {
+			return nil, err
+		}
+		backups = append(backups, backup)
+	}
+	sort.Slice(backups, func(i, j int) bool {
+		return backups[i].FinishLocalTime.After(backups[j].FinishLocalTime)
+	})
+	return backups, nil
+}
+
+func BackupMeta(folder storage.Folder, name string) (archive.Backup, error) {
+	backup := internal.NewBackup(folder, name)
+	var sentinel archive.Backup
+	err := backup.FetchSentinel(&sentinel)
+	if err != nil {
+		return archive.Backup{}, fmt.Errorf("can not fetch stream sentinel: %w", err)
+	}
+	if sentinel.BackupName == "" {
+		sentinel.BackupName = name
+	}
+	return sentinel, nil
+}
+
+// BackupNamesFromBackupTimes forms list of backup names from BackupTime
+func BackupNamesFromBackupTimes(backups []internal.BackupTime) []string {
+	names := make([]string, 0, len(backups))
+	for _, b := range backups {
+		names = append(names, b.BackupName)
+	}
+	return names
+}
+
+// DeleteGarbage purges given garbage keys
+func DeleteGarbage(folder storage.Folder, garbage []string) error {
```

Comment:
nit: looks like 'DeletePrefixes' function somewhere in common lib.

Reply:
This is copy of 'func (sp *StoragePurger) DeleteGarbage(garbage []string) error {'
I will add comment here